packages/grafana-ui/src/components/PanelOptions/OptionsUIBuilder.test.ts

```diff
@@ -0,0 +1,28 @@
+import { OptionsGroupUIBuilder } from './OptionsUIBuilder';
+import { SingleStatBaseOptions } from '../SingleStatShared/SingleStatBaseOptions';
+interface GaugeOptions extends SingleStatBaseOptions {
+  showThresholdLabels: boolean;
+  showThresholdMarkers: boolean;
+}
+
+describe('OptionsUIBuilder', () => {
+    it('allows group as root', () => {
+      const builder = new OptionsGroupUIBuilder<GaugeOptions>();
+
+      const schema = builder
+          .addGroup({})
+            .addBooleanEditor('showThresholdLabels')
+            .addNestedOptionsGroup('fieldOptions', {})
```

Comment:
@torkelo I have came up with this "fake grouping" for nested options. Type inferring based on property name works, yay, both in flat and nested scenario. But, see comments in the code below. Can't make the type after the group is ended (.groupEnd) be correct(it holds to nested option type down the rowad). Any ideas how the implementation of `endGroup` could be improved to handle that? Feels like I'm missing sth obvious here. cc @ryantxu